EpiAware/src/epimodel.jl

```diff
@@ -28,14 +28,15 @@ struct EpiModel{T<:Real} <: AbstractEpiModel
     cluster_coeff::T
     len_gen_int::Integer #length(gen_int) just to save recalc
     len_delay_int::Integer #length(delay_int) just to save recalc
+    time_horizon::Integer
 
     #Inner constructors for EpiModel object
-    function EpiModel(gen_int, delay_int, cluster_coeff, time_horizon)
+    function EpiModel(gen_int, delay_int, cluster_coeff, time_horizon::Integer)
         @assert all(gen_int .>= 0) "Generation interval must be non-negative"
         @assert all(delay_int .>= 0) "Delay interval must be non-negative"
         @assert sum(gen_int) ≈ 1 "Generation interval must sum to 1"
         @assert sum(delay_int) ≈ 1 "Delay interval must sum to 1"
-        #construct observation delay kernel
+
         K = generate_observation_kernel(delay_int, time_horizon)
 
         new{eltype(gen_int)}(
```

Comment:
can you explain to me like I am 15 what this bit does?

Reply:
Structs come with a default constructor. You can also define inner and outer constructors. Inner constuctors come with a reserved keyword `new` and the chance to give a type for parametric typed structs.

The docs are [here](https://docs.julialang.org/en/v1/manual/constructors/)